### PR TITLE
ElasticsearchV2 - Added Support for v8.x 

### PR DESCRIPTION
**Pull request by ShacharKidor**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-10581](https://jira-dc.paloaltonetworks.com/browse/CIAC-10581).

## Related Dockerfiles PR
relates: [pull/32947](https://github.com/demisto/dockerfiles/pull/32947).

## Related Content PR
relates: [pull/36849](https://github.com/demisto/content/pull/36849).

## Description
- Added support for elasticsearch v8. 
- Maintain BC for elasticsearch v >= 7 and OpenSearch. 
- Added a new command - 'es-get-indices-statistics' - to get info and statistics of all existing indices in the elasticsearch server. 
- Fixed the Elasticsearch_v2_test-v8 TPB and added v8 instance.
- Added a TPB for testing the integration commands on both elasticsearch v7 and v8 instances: 
Elasticsearch_v2_test-elastic-v7-v8_Tue_Oct_08_2024

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml [hunk 1]
```diff
@@ -289,6 +290,40 @@ script:
...
+    - contextPath: Elasticsearch.IndexStatistics.Status
+      description: The Status of the index.
+      type: String
+    - contextPath: Elasticsearch.IndexStatistics.Health
+      description: The Health status of the index.
+      type: String
+    - contextPath: Elasticsearch.IndexStatistics.UUID
+      description: The UUID of the index.
+      type: String
+    - contextPath: Elasticsearch.IndexStatistics.Documents Count
```
Comment (ShirleyDenkberg) [inline Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml:321]: Is a period or underscore missing in the context path? Or is there an extra space in Documents Count?

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml [hunk 2]
```diff
@@ -289,6 +290,40 @@ script:
...
+      description: The amount of documents that are indexed in the index.
+      type: Number
+    - contextPath: Elasticsearch.IndexStatistics.Documents Deleted
+      description: The amount of documents that were deleted from the index.
```
Comment (ShirleyDenkberg) [inline Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml:325]:
```suggestion
      description: The number of documents that were deleted from the index.
```

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/README.md [hunk 3]
```diff
@@ -20,7 +21,7 @@ The permissions required to use this integration depends on which operations you
     | Username for server login | Provide Username \+ Passoword instead of API key \+ API ID | False |
     | Trust any certificate (not secure) |  | False |
     | Use system proxy settings |  | False |
-    | Client type | In some hosted ElasticSearch environments, the standard ElasticSearch client is not supported. If you encounter any related client issues, please consider using the OpenSearch client type. | False |
+    | Client type | For Elasticsearch version 7 and below select 'Elasticsearch', for Elasticsearch server version 8 select 'Elasticsearch_v8'. In some hosted ElasticSearch environments, the standard ElasticSearch client is not supported. If you encounter any related client issues, please consider using the OpenSearch client type. | False |
```
Comment (ShirleyDenkberg) [inline Packs/Elasticsearch/Integrations/Elasticsearch_v2/README.md:24]:
```suggestion
    | Client type | For Elasticsearch version 7 and below, select 'Elasticsearch'. For Elasticsearch server version 8, select 'Elasticsearch_v8'. In some hosted Elasticsearch environments, the standard Elasticsearch client is not supported. If you encounter any related client issues, consider using the OpenSearch client type. | False |
```

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/README.md [hunk 4]
```diff
@@ -245,4 +245,31 @@ There are no input arguments for this command.
...
+#### Context Output
+
+| **Path** | **Type** | **Description** |
+| --- | --- | --- |
+| Elasticsearch.IndexStatistics.Name | String | The Name of the index. |
+| Elasticsearch.IndexStatistics.Status | String | The Status of the index. |
+| Elasticsearch.IndexStatistics.Health | String | The Health status of the index. |
+| Elasticsearch.IndexStatistics.UUID | String | The UUID of the index. |
+| Elasticsearch.IndexStatistics.Documents Count | Number | The amount of documents that are indexed in the index. |
+| Elasticsearch.IndexStatistics.Documents Deleted | Number | The amount of documents that were deleted from the index. |
```
Comment (ShirleyDenkberg) [inline Packs/Elasticsearch/Integrations/Elasticsearch_v2/README.md:275]:
```suggestion
| Elasticsearch.IndexStatistics.Name | String | The name of the index. |
| Elasticsearch.IndexStatistics.Status | String | The status of the index. |
| Elasticsearch.IndexStatistics.Health | String | The health status of the index. |
| Elasticsearch.IndexStatistics.UUID | String | The UUID of the index. |
| Elasticsearch.IndexStatistics.Documents Count | Number | The number of documents that are indexed in the index. |
| Elasticsearch.IndexStatistics.Documents Deleted | Number | The number of documents that were deleted from the index. |
```

**Comment by ShirleyDenkberg** [comment]
@sapirshuker @yuvalbenshalom Doc review completed.

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml (hunk 1 above)
Comment (ShacharKidor) [reply Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2.yml:321]: This is an extra space, I'll fix that, thank you!

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2_test.py [hunk 5]
```diff
@@ -1103,3 +1128,56 @@ def test_key_not_found():
...
+                                         ]
+                              )],
+                         ids=[
+                             "Test get indices statistics with a limit and all_results=True",
+                             "Test get indices statistics with a limit and all_results=False"]
+                         )
+def test_get_indices_statistics_command(mocker, limit, all_results, expected_context):
+    """
+    Given
+      - elastic search client
```
Comment (yuvalbenshalom) [inline Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2_test.py:1165]: the "given" should, in most cases, give explanation about the pre setup. 
so it is not the client it self, but the variables that are configured in ti an teir meaning.
In general, when using parametrize, it is better to add the docstring per case.

**Review by yuvalbenshalom — APPROVED** [review]
looks good. see my comment.

File: Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2_test.py (hunk 5 above)
Comment (ShacharKidor) [reply Packs/Elasticsearch/Integrations/Elasticsearch_v2/Elasticsearch_v2_test.py:1165]: Will ad it, thanks!

**Review by amshamah419 — APPROVED** [review]

**Pull request merged**